Self-Consistent-Field/vcd.py

```diff
@@ -0,0 +1,874 @@
+conv_kmmol *= 1
+# Multiply by (Na pi / 3 c^2) in a.u. (Na = mol^-1; c = Eh / alpha^2 me) to get to units of a0/mol
+conv_kmmol *= psi_na * np.pi * (1 / 3) * psi_alpha**2
+# Convert to units of km/mol
+conv_kmmol *= psi_bohr2m * (1 / 1000)
+#
+conv_ir_DAamu2kmmol = conv_kmmol                                                                                # (D/A)^2/amu -> km/mol
+
+
+# Specify Molecule
+#mol = psi4.geometry("""
+#    O            0.000000000000     0.000000000000    -0.075791843897
+#    H            0.000000000000    -0.866811832375     0.601435781623
+#    H            0.000000000000     0.866811832375     0.601435781623
+#    symmetry c1
+#    units ang
+#    noreorient
+#""")
+#mol = psi4.geometry("""
+#    O     0.000000000000     0.000000000000    -0.134503695264
+#    H     0.000000000000    -1.684916670000     1.067335684736
+#    H     0.000000000000     1.684916670000     1.067335684736
+#    symmetry c1
+#    units bohr
+#    noreorient
+#""")
+#mol = psi4.geometry("""
+#    O           -0.0000000000        1.3127774000       -0.1092016150
+#    O           -0.0000000000       -1.3127774000       -0.1092016150
+#    H            1.4385007800        1.7079033100        0.8736129170
+#    H           -1.4385007800       -1.7079033100        0.8736129170
+#    symmetry c1
+#    units bohr
+#    noreorient
+#""")
+mol = psi4.geometry("""
+    O       0.0000000000        1.3192641900       -0.0952542913
+    O      -0.0000000000       -1.3192641900       -0.0952542913
+    H       1.6464858700        1.6841036400        0.7620343300
+    H      -1.6464858700       -1.6841036400        0.7620343300
+    symmetry c1
+    units bohr
+    noreorient
+    no_com
+""")
+
+psi4.core.set_active_molecule(mol)
+
+geom = psi4.core.Matrix.to_array(mol.geometry())
+
+# Set Psi4 Options
+options = {'BASIS':'STO-3G',
+           'SCF_TYPE':'PK',
+           'E_CONVERGENCE':1e-10,
+           'D_CONVERGENCE':1e-10}
+psi4.set_options(options)
+
+# Perform SCF Energy Calculation
+rhf_e, wfn = psi4.frequency('SCF', return_wfn=True)
+
+# Relevant Variables
+natoms = mol.natom()
+nmo = wfn.nmo()
+nocc = wfn.doccpi()[0]
+nvir = nmo - nocc
+
+# MO Coefficients
+C = wfn.Ca_subset("AO", "ALL")
+npC = psi4.core.Matrix.to_array(C)
+
+# Integral generation from Psi4's MintsHelper
+mints = psi4.core.MintsHelper(wfn.basisset())
+
+# Build T, V, and S
+T = mints.ao_kinetic()
+npT = psi4.core.Matrix.to_array(T)
+V = mints.ao_potential()
+npV = psi4.core.Matrix.to_array(V)
+S = mints.ao_overlap()
+npS = psi4.core.Matrix.to_array(S)
+
+# Transform S to MO Basis
+npS_mo = 2.0 * np.einsum('uj,vi,uv', npC, npC, npS, optimize=True)
+
+# Build AO Dipole Integrals
+MU = mints.ao_dipole()
+AM = mints.ao_angular_momentum()
+npMU = []
+npAM = []
+npMU_mo = []
+npAM_mo = []
+for cart in range(3):
+    npMU.append(psi4.core.Matrix.to_array(MU[cart]))
+    npAM.append(psi4.core.Matrix.to_array(AM[cart]))
+    npAM[cart] *= -0.5
```

Comment:
I am not sure about the sign here. In the computation of ECD rotatory strengths I use 1/2 for the Bohr magneton in atomic units: https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/response/scf_response.py#L368-L370

Reply:
You're right. I corrected it to a positive factor of ½. I also had to correct a minus sign on the RHS of the CPHF equations with respect to the magnetic perturbation where these integrals appeared later.